src/moon.ts

```diff
@@ -0,0 +1,41 @@
+/// <reference types="../types/suncalc" />
+
+import {getMoonIllumination, getMoonTimes} from 'suncalc';
+import {isAfter, isBefore} from 'date-fns';
+
+export function moonIsVisible(lat: number, long: number, date: Date, isUTC: boolean): boolean {
+  const times = getMoonTimes(date, lat, long, isUTC);
+
+  // ez short circuit: if the moon is always up or always down that day, whatever
+  if (times.alwaysUp) {
+    return true;
+  } else if (times.alwaysDown) {
+    return false;
+  }
+
+  // TODO: Holy shit there must be a better way to do this
+  if (isAfter(times.rise, times.set)) {
+    // If the moonrise on this date is AFTER moonset, we need to make sure the time is either
+    // BEFORE both or AFTER both
+    if (isBefore(date, times.rise) && isBefore(date, times.set)) {
```

Comment:
Might end up being more readable to assign these bools to local vars like `isBeforeMoonRise`, `isBeforeMoonSet` etc